python/taulabs/uavo.py

```diff
@@ -269,10 +346,25 @@ class tmpClass(namedtuple(name, tuple_fields), UAVTupleClass):
     # child classes don't get a dict / keep all their namedtuple goodness
     tuple_class = type(name, (tmpClass,), { "__slots__" : () })
 
+    # Set the default values for the constructor
+    defaults = [name, 0, uavo_id]
+    defaults.extend(f['defaultvalue'] for f in fields)
+    tuple_class.__new__.__defaults__ = tuple(defaults)
+
+    # Add enums
+    for field in fields:
+        if field['type'] == 'enum':
+            enum = field['options']
+            mapping = dict((key, value) for key, value in enum.iteritems())
+            reverse_mapping = dict((value, key) for key, value in enum.iteritems())
+            setattr(tuple_class, 'ENUM_' + field['name'], mapping)
+            setattr(tuple_class, 'ENUMR_' + field['name'], reverse_mapping)
```

Comment:
Instead of using a two dicts, we could just use one dict that contains the mapping from integer to string and vice versa. May be easier to use. I.e., `ENUM_FlightMode[0]` would return a string while `ENUM_FlightMode['Manual']` would return an int. Thoughts?